examples/pennylane_universal_classifier.py

```diff
@@ -0,0 +1,491 @@
+leads to its projection as either the state 0 or 1. We can compute the
+fidelity (or closeness) of the output state to the class label making
+the output state jump to either :math:`| 0 \rangle` or
+:math:`|1\rangle`. By repeating this process several times, we can
+compute the probability or overlap of our output to both labels and
+assign a class based on the label our output has a higher overlap. This
+is much like having a set of ouput neurons and selecting the one which
+has the highest value as the label.
+
+In ``pennylane``, we can define an observable (the expected output
+label) and make a circuit to return the fidelity using the ``Hermitian``
+operator. We can then define the cost function as the sum of the
+fidelities for all the data points after passing throuh the circuit
+and optimize the parameters :math:`(\vec \theta)` to minimize the cost.
+
+.. math::
+
+
+   \texttt{Cost} = \sum_{\texttt{data points}} (1 - \texttt{fidelity}(\psi_{\texttt{output}}(\vec x, \vec \theta), \psi_{\texttt{label}}))
+
+Now, we can use our favorite optimizer to maximize the sum of the
+fidelities over all data points (or batches of datapoints) and find the
+optimal weights for classification. Gradient based optimizers such as
+Adam can be used if we have a good model of the circuit and how noise
+might affect it. Or, we can use some gradient-free method such as LBFGS
+to evaluate the gradient and find the optimal weights where we can treat
+the quantum circuit as a black-box and the gradients are computed
+numerically using a fixed number of function evalutaions and iterations.
+
+Multiple qubits, entanglement and Deep Neural Networks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Universal Approximation Theorem declares a single hidden layered
+neural network to be capable of approximating any function to arbitrary
+accuracy. But in practice, it might require a large number of neurons in
+the single hidden layer and here is where Deep Neural Networks come into
+action. Deep Neural Networks proved to be better in practice and we have
+some intuitive idea why, read “Why does deep and cheap learning work so
+well?” by Henry W. Lin, Max Tegmark (MIT) and David Rolnick (2016).
+
+Adri´an P´erez-Salinas et al. (2019) describe that in their approach the
+“layers” :math:`L_i(\vec \theta_i, \vec x )` are analogous to the size
+of the intermediate hidden layer of the neural network. And what counts
+for deep (multiple layers of the neural network) relates to the number
+of qubits. So, multiple qubits with entanglement between them could
+provide some quantum advantage over classical neural networks. But here,
+we will only implement a single qubit classifier.
+
+.. figure:: ../../examples/figures/universal_dnn.png
+   :scale: 35%
+   :alt: DNN
+
+
+References
+==========
+
+[1] Pérez-Salinas, Adrián, et al. “Data re-uploading for a universal
+quantum classifier.” arXiv preprint arXiv:1907.02085 (2019).
+
```

Comment:
Might make more sense in terms of the flow to place the references at the end of the tutorial